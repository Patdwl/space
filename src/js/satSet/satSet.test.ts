import { defaultSat, keepTrackApiStubs } from '../api/apiMocks';
import { keepTrackApi } from '../api/keepTrackApi';
import { KeepTrackPrograms, SatCruncherMessage, SatObject } from '../api/keepTrackTypes';
import { SpaceObjectType } from '../api/SpaceObjectType';
import * as satSet from '../satSet/satSet';
import { cruncherDotsManagerInteraction, cruncherExtraUpdate, onCruncherReady, parseGetVariables, satCruncherOnMessage } from './catalogSupport/cruncherInteractions';
import {
  getIdFromEci,
  getIdFromIntlDes,
  getIdFromObjNum,
  getIdFromStarName,
  getSat,
  getSatExtraOnly,
  getSatInSun,
  getSatInView,
  getSatPosOnly,
  getSatVel,
  getScreenCoords,
  getSensorFromSensorName,
} from './catalogSupport/getters';

keepTrackApi.programs = <KeepTrackPrograms>(<unknown>{ ...keepTrackApi.programs, ...keepTrackApiStubs.programs });

const defaultSatCruncherMessage: SatCruncherMessage = {
  data: {
    satPos: [0, 0, 0],
    satVel: [0, 0, 0],
    satInSun: [1],
    satInView: [1],
    sensorMarkerArray: [1],
    satId: 0,
    extraData: JSON.stringify([
      {
        inclination: 0,
        eccentricity: 0,
        raan: 0,
        argPe: 0,
        meanMotion: 0,
        semiMajorAxis: 0,
        semiMinorAxis: 0,
        apogee: 0,
        perigee: 0,
        period: 0,
        TLE1: '1234567890',
        TLE2: '1234567890',
      },
    ]),
    extraUpdate: true,
  },
};

describe('satSet.init', () => {
  test('0', async () => {
    await satSet.init(keepTrackApi.programs.satSet.satCruncher);
  });

  test('1', async () => {
    await satSet.init(keepTrackApi.programs.satSet.satCruncher);
  });
});

describe('satSet.parseGetVariables', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });
  test('0', () => {
    let result: any = parseGetVariables();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSat', () => {
  it('should return extra info if gotExtraData is true', () => {
    satSet.init(keepTrackApi.programs.satSet.satCruncher);
    keepTrackApi.programs.satSet.gotExtraData = true;
    let result: any = getSat(0);
    expect(result.velocity).not.toBe(false);
  });
  test('0', () => {
    let result: any = getSat(-5.48);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getSat(-100);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = getSat(0);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = getSat(100);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = getSat(1);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = getSat(-Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.selectSat', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
    keepTrackApi.programs.colorSchemeManager.currentColorScheme = keepTrackApi.programs.colorSchemeManager.default;
    keepTrackApi.programs.colorSchemeManager.colorBufferOneTime = true;
  });
  test('0', () => {
    let result: any = satSet.selectSat(-1);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = satSet.selectSat(1);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = satSet.selectSat(0);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = satSet.selectSat(100);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = satSet.selectSat(150);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = satSet.selectSat(Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSatPosOnly', () => {
  test('0', () => {
    let result: any = getSatPosOnly(100);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getSatPosOnly(-5.48);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = getSatPosOnly(1);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = getSatPosOnly(0);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = getSatPosOnly(-100);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = getSatPosOnly(NaN);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getIdFromEci', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });
  test('0', () => {
    let result: any = getIdFromEci({ x: 1, y: 520, z: 350 });
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getIdFromEci({ x: 350, y: 550, z: 90 });
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = getIdFromEci({ x: 4, y: 30, z: 400 });
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = getIdFromEci({ x: 90, y: 350, z: 70 });
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = getIdFromEci({ x: 520, y: 550, z: 4 });
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = getIdFromEci({ x: -Infinity, y: -Infinity, z: -Infinity });
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getIdFromObjNum', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });
  test('0', () => {
    let result: any = getIdFromObjNum(25544);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getIdFromObjNum(-1);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSatInView', () => {
  test('0', () => {
    let result: any = getSatInView();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSatInSun', () => {
  test('0', () => {
    let result: any = getSatInSun();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSatVel', () => {
  test('0', () => {
    let result: any = getSatVel();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.resetSatInView', () => {
  test('0', () => {
    let result: any = satSet.resetSatInView();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.resetSatInSun', () => {
  test('0', () => {
    let result: any = satSet.resetSatInSun();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSatExtraOnly', () => {
  beforeEach(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });
  test('0', () => {
    let result: any = getSatExtraOnly(0);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getSatExtraOnly(-100);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = getSatExtraOnly(1);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = getSatExtraOnly(-5.48);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = getSatExtraOnly(100);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = getSatExtraOnly(Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.setSat', () => {
  test('0', () => {
    let result: any = satSet.setSat(10, defaultSat);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = satSet.setSat(0, defaultSat);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = satSet.setSat(48, defaultSat);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = satSet.setSat(-100, defaultSat);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = satSet.setSat(-1100, defaultSat);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = satSet.setSat(Infinity, defaultSat);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.onCruncherReady', () => {
  test('0', () => {
    let result: any = onCruncherReady();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getScreenCoords', () => {
  test('0', () => {
    let param2: any = new Float32Array([10.23, 0.5, -1.0, -1.0, 0.0, -1.0, 1.0, -1.0, -29.45, 1.0]);
    let param3: any = new Float32Array([1.0, 1.0, 0.0]);
    let result: any = getScreenCoords(-100, param2, param3, { x: 320, y: 4, z: 320 });
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let param3: any = new Float32Array([-1.0, -0.5, 10.23]);
    let result: any = getScreenCoords(1, [1.0, 0.0, 0.5, 0.5, 10.0, -29.45, 0.5, 10.0, 0.5, -0.5, -29.45, 10.23, 10.23, -1.0, 10.0, -29.45], param3, { x: 70, y: 520, z: 400 });
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let param3: any = new Float32Array([-1.0, 10.23, 0.5]);
    let result: any = getScreenCoords(-100, [10.23, 10.0, 1.0, 0.5, 0.5, 1.0, 0.0, 10.0, -0.5, 10.0, 1.0, -29.45, -29.45, -1.0, 10.23, 0.5], param3, { x: 70, y: 350, z: 400 });
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let param3: any = new Float32Array([10.0, -0.5, 0.0]);
    let result: any = getScreenCoords(-5.48, [-0.5, 1.0, 1.0, -1.0, 1.0, -1.0, -29.45, -29.45, -29.45, -0.5, -29.45, -0.5, -0.5, -0.5, 1.0, -1.0], param3, {
      x: 30,
      y: 410,
      z: 400,
    });
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = getScreenCoords(
      100,
      [10.23, 0.0, 10.23, 0.5, 1.0, 0.0, -0.5, 0.0, 10.23, 10.0, 0.0, -1.0, 0.5, -29.45, 0.0, -1.0],
      [-1.0, -1.0, -1.0, -0.5, -0.5, 1.0, -1.0, -1.0, 1.0, -0.5, 10.0, 10.23, -1.0, -1.0, -29.45, -0.5],
      {
        x: 50,
        y: 4,
        z: 4,
      }
    );
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let param2: any = new Float32Array([]);
    let result: any = getScreenCoords(NaN, param2, [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], { x: NaN, y: NaN, z: NaN });
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.convertIdArrayToSatnumArray', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });
  test('0', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([10, 10, 64]);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([32, 256, 16]);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([64, 16, 16]);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([10, 10, 0]);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([0, 256, 0]);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = satSet.convertIdArrayToSatnumArray([]);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.convertSatnumArrayToIdArray', () => {
  test('0', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([-5.48, -100, 100, 1, -5.48]);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([100, -5.48, 1, 0]);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([100, -5.48, 0, 100, -100]);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([-100, 100, 1, 100]);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([100, -5.48, -5.48, -5.48]);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = satSet.convertSatnumArrayToIdArray([]);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getIdFromIntlDes', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = getIdFromIntlDes('1998-AB');
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getIdFromIntlDes('2020-A');
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getIdFromStarName', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = getIdFromStarName('test');
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getIdFromStarName('2020-A');
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.getSensorFromSensorName', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = getSensorFromSensorName('test');
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = getSensorFromSensorName('2020-A');
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.mergeSat', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = satSet.mergeSat(<SatObject>(<unknown>{ sccNum: '25544', type: SpaceObjectType.PAYLOAD }));
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('satSet.addSatExtraFunctions', () => {
  test('0', () => {
    const fakeSatSet = <any>keepTrackApiStubs.programs.satSet;
    delete fakeSatSet.satData[0].isInSun;
    satSet.replaceSatSet(fakeSatSet);
    let result: any = satSet.addSatExtraFunctions(0);
    expect(result).toMatchSnapshot();
    expect(() => satSet.satSet.satData[0].isInSun()).not.toThrow();
  });
});

describe('satSet.satCruncherOnMessage', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = satCruncherOnMessage(defaultSatCruncherMessage);
    expect(result).toMatchSnapshot();
  });
});

describe('satSet.cruncherExtraUpdate', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = cruncherExtraUpdate(defaultSatCruncherMessage);
    expect(result).toMatchSnapshot();
  });
});

describe('satSet.cruncherDotsManagerInteraction', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = cruncherDotsManagerInteraction(defaultSatCruncherMessage);
    expect(result).toMatchSnapshot();
  });
});

describe('satSet.satCruncherOnMessage', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    let result: any = satCruncherOnMessage(defaultSatCruncherMessage);
    expect(result).toMatchSnapshot();
  });
});

describe('satSet.insertNewAnalystSatellite', () => {
  beforeAll(() => {
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    satSet.insertNewAnalystSatellite(defaultSat.TLE1, defaultSat.TLE2, 0);
  });
});

describe('satSet.setHover', () => {
  beforeAll(() => {
    keepTrackApi.programs.colorSchemeManager.currentColorScheme = keepTrackApi.programs.colorSchemeManager.default;
    satSet.replaceSatSet(keepTrackApiStubs.programs.satSet);
  });

  test('0', () => {
    keepTrackApi.programs.colorSchemeManager.currentColorScheme = keepTrackApi.programs.colorSchemeManager.default;
    let result: any = satSet.setHover(0);
    expect(result).toMatchSnapshot();
  });
});
