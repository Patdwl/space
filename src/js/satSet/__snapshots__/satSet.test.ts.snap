// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`satSet.addSatExtraFunctions 0 1`] = `undefined`;

exports[`satSet.convertIdArrayToSatnumArray 0 1`] = `
Array [
  "00005",
  "00005",
  "00005",
]
`;

exports[`satSet.convertIdArrayToSatnumArray 1 1`] = `
Array [
  "00005",
  "00005",
  "00005",
]
`;

exports[`satSet.convertIdArrayToSatnumArray 2 1`] = `
Array [
  "00005",
  "00005",
  "00005",
]
`;

exports[`satSet.convertIdArrayToSatnumArray 3 1`] = `
Array [
  "00005",
  "00005",
  "00005",
]
`;

exports[`satSet.convertIdArrayToSatnumArray 4 1`] = `
Array [
  "00005",
  "00005",
  "00005",
]
`;

exports[`satSet.convertIdArrayToSatnumArray 5 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 0 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 1 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 2 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 3 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 4 1`] = `Array []`;

exports[`satSet.convertSatnumArrayToIdArray 5 1`] = `Array []`;

exports[`satSet.cruncherDotsManagerInteraction 0 1`] = `undefined`;

exports[`satSet.cruncherExtraUpdate 0 1`] = `undefined`;

exports[`satSet.getIdFromEci 0 1`] = `-1`;

exports[`satSet.getIdFromEci 1 1`] = `-1`;

exports[`satSet.getIdFromEci 2 1`] = `-1`;

exports[`satSet.getIdFromEci 3 1`] = `-1`;

exports[`satSet.getIdFromEci 4 1`] = `-1`;

exports[`satSet.getIdFromEci 5 1`] = `-1`;

exports[`satSet.getIdFromIntlDes 0 1`] = `5`;

exports[`satSet.getIdFromIntlDes 1 1`] = `null`;

exports[`satSet.getIdFromObjNum 0 1`] = `null`;

exports[`satSet.getIdFromObjNum 1 1`] = `null`;

exports[`satSet.getIdFromStarName 0 1`] = `null`;

exports[`satSet.getIdFromStarName 1 1`] = `null`;

exports[`satSet.getSat 0 1`] = `null`;

exports[`satSet.getSat 1 1`] = `null`;

exports[`satSet.getSat 2 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 0,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSat 3 1`] = `null`;

exports[`satSet.getSat 4 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 1,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": undefined,
    "y": undefined,
    "z": undefined,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSat 5 1`] = `null`;

exports[`satSet.getSatExtraOnly 0 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 0,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSatExtraOnly 1 1`] = `null`;

exports[`satSet.getSatExtraOnly 2 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 1,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": undefined,
    "y": undefined,
    "z": undefined,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSatExtraOnly 3 1`] = `null`;

exports[`satSet.getSatExtraOnly 4 1`] = `null`;

exports[`satSet.getSatExtraOnly 5 1`] = `null`;

exports[`satSet.getSatInSun 0 1`] = `
Float32Array [
  1,
  0,
  0,
]
`;

exports[`satSet.getSatInView 0 1`] = `
Float32Array [
  1,
  0,
  0,
]
`;

exports[`satSet.getSatPosOnly 0 1`] = `null`;

exports[`satSet.getSatPosOnly 1 1`] = `null`;

exports[`satSet.getSatPosOnly 2 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 1,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": undefined,
    "y": undefined,
    "z": undefined,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSatPosOnly 3 1`] = `
Object {
  "TLE1": "1 25544U 98067A   21203.40407588  .00003453  00000-0  71172-4 0  9991",
  "TLE2": "2 25544  51.6423 168.5744 0001475 184.3976 313.3642 15.48839820294053",
  "active": true,
  "apogee": 184.3976,
  "argPe": 0,
  "country": "ISS",
  "dryMass": "",
  "eccentricity": 0.00003453,
  "getAltitude": [Function],
  "getDirection": [Function],
  "getTEARR": [Function],
  "id": 0,
  "inView": 1,
  "inclination": 51.6423,
  "intlDes": "1998-AEF",
  "isInSun": [Function],
  "launchMass": "",
  "launchSite": "TTMTR",
  "launchVehicle": "Proton-K",
  "lifetime": 30,
  "lon": 168.5744,
  "manufacturer": "Boeing Satellite Systems (prime)/Multinational",
  "meanMotion": 15.48839820294053,
  "mission": "Final size of a Boeing 747; first component.",
  "name": "ISS (ZARYA)",
  "owner": "National Aeronautics and Space Administration (NASA)/Multinational",
  "perigee": 1475,
  "period": 313.3642,
  "position": Object {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "power": "",
  "purpose": "Space Science",
  "raan": 0,
  "rcs": "99.0524",
  "sccNum": "00005",
  "semimajorAxis": 21203.40407588,
  "setRAE": [Function],
  "static": false,
  "staticNum": null,
  "type": 1,
  "user": "Government",
  "velocity": Object {
    "total": 0,
    "x": 0,
    "y": 0,
    "z": 0,
  },
}
`;

exports[`satSet.getSatPosOnly 4 1`] = `null`;

exports[`satSet.getSatPosOnly 5 1`] = `null`;

exports[`satSet.getSatVel 0 1`] = `
Float32Array [
  0,
  0,
  0,
]
`;

exports[`satSet.getScreenCoords 0 1`] = `
Object {
  "error": true,
  "x": NaN,
  "y": NaN,
  "z": NaN,
}
`;

exports[`satSet.getScreenCoords 1 1`] = `
Object {
  "error": true,
  "x": NaN,
  "y": NaN,
  "z": NaN,
}
`;

exports[`satSet.getScreenCoords 2 1`] = `
Object {
  "error": true,
  "x": NaN,
  "y": NaN,
  "z": NaN,
}
`;

exports[`satSet.getScreenCoords 3 1`] = `
Object {
  "error": true,
  "x": NaN,
  "y": NaN,
  "z": NaN,
}
`;

exports[`satSet.getScreenCoords 4 1`] = `
Object {
  "error": true,
  "x": -125134.46216952907,
  "y": 55817.85373904889,
  "z": -63.31606182443326,
}
`;

exports[`satSet.getScreenCoords 5 1`] = `
Object {
  "error": true,
  "x": NaN,
  "y": NaN,
  "z": NaN,
}
`;

exports[`satSet.getSensorFromSensorName 0 1`] = `3`;

exports[`satSet.getSensorFromSensorName 1 1`] = `-1`;

exports[`satSet.mergeSat 0 1`] = `undefined`;

exports[`satSet.onCruncherReady 0 1`] = `undefined`;

exports[`satSet.parseGetVariables 0 1`] = `undefined`;

exports[`satSet.resetSatInSun 0 1`] = `undefined`;

exports[`satSet.resetSatInView 0 1`] = `undefined`;

exports[`satSet.satCruncherOnMessage 0 1`] = `undefined`;

exports[`satSet.satCruncherOnMessage 0 2`] = `undefined`;

exports[`satSet.selectSat 0 1`] = `undefined`;

exports[`satSet.selectSat 1 1`] = `undefined`;

exports[`satSet.selectSat 2 1`] = `undefined`;

exports[`satSet.selectSat 3 1`] = `undefined`;

exports[`satSet.selectSat 4 1`] = `undefined`;

exports[`satSet.selectSat 5 1`] = `undefined`;

exports[`satSet.setHover 0 1`] = `undefined`;

exports[`satSet.setSat 0 1`] = `undefined`;

exports[`satSet.setSat 1 1`] = `undefined`;

exports[`satSet.setSat 2 1`] = `undefined`;

exports[`satSet.setSat 3 1`] = `undefined`;

exports[`satSet.setSat 4 1`] = `undefined`;

exports[`satSet.setSat 5 1`] = `undefined`;
